Adding ID Token roles mapping

vakninr opened the pull request:
* Adding fallback support to read roles from ID Token which is JWT format
* Taking the roles from the JWT based on the OIDC_ROLE_PATH_IN_RETURN
* Added test to validate the role mapping

GDay commented on back/user_auth/views.py:
```diff
@@ -322,26 +323,46 @@ def __sync_user(self, user_info):
+            return {}
+        
+        try:
+            claims = jwt.decode(id_token,key='secret',algorithms=["HS256"])
```
I am not super familiar with OIDC, but can we safely assume that this key is always "secret"?

Just to be on the safe side, we might want to use this to do the validation and extracting info: https://pyjwt.readthedocs.io/en/latest/usage.html#oidc-login-flow

vakninr replied:
Sure I will fix the flake8,black was not sure how to run it. Regarding the secret I will validate, but even when you decrypt jwt in jwt.io website you don't need really the secret. The secret is usually used to do validation on the JWT that it is valid and properly signed. But here I'm not concerned about security and don't really care of the signing validation just about extracting the claims out of the token.
The validation is already done using the OIDC protocols and the auth module...here we just checking the token info

GDay reviewed (changes requested):
Thanks a lot for the PR! Really appreciate it!

Could you fix flake8/black please? (see error from Github action)

vakninr commented:
I updated the PR, Seems like it passed the all the isort,black and flake8. (At least locally)
I modified the function that parse the claims by changing the secret to an option that no validation will be needed. (As I wrote above it is not interesting here, we just want the claims).

GDay reviewed (approved):
LGTM! Thanks!
